x/muxer/mapping.go

```diff
@@ -0,0 +1,300 @@
+// Copyright 2016-2023, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package muxer
+
+import (
+	"strings"
+
+	"github.com/pulumi/pulumi/pkg/v3/codegen/schema"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/contract"
+)
+
+type ComputedMapping struct {
+	// mapping is an implementation detail.
+	//
+	// Right now, we want to expose the same information to the user, but that doesn't
+	// need to be true forever.
+	mapping
+}
+
+func Mapping(schemas []schema.PackageSpec) (ComputedMapping, schema.PackageSpec, error) {
+	// TODO Insert sanity checks and return an error on conflicts.  For example right
```

Comment:
Very much worth doing especially for basic things like both providers define the same resource. I think we have to do it. Please let's link a ticket and let's do it soon, I can help perhaps.

Reply:
https://github.com/pulumi/pulumi-terraform-bridge/issues/949